cpp/src/utilities/prefetch.cpp

```diff
@@ -32,15 +32,18 @@ prefetch_config& prefetch_config::instance()
   return instance;
 }
 
-bool prefetch_config::get(std::string_view key)
+bool prefetch_config::get(std::string_view key) const
 {
   // Default to not prefetching
-  if (config_values.find(key.data()) == config_values.end()) {
-    return (config_values[key.data()] = false);
-  }
-  return config_values[key.data()];
+  if (config_values.find(key.data()) == config_values.end()) { return false; }
+  return config_values.get(key.data());
```

Comment:
don't we need to protect the `::get` as well with the same mutex? If we go down the mutex route?

Reply:
The `get` does not modify map. It just  access. The previous bracket access `[]` actually modifies map if the key doesn't exist.

Reply:
accessing an STL container and mutating it at the same time (which could happen here) is thread safe?

Reply:
Yes it (`map::operator[]`) is not, so it is removed now, and replaced by `map::get` to be thread-safe.

Reply:
Okay, I was wrong. `get` will not cause corruption but still not thread-safe.

Reply:
right but what about this scenario:

1. Thread 1 goes to `set` and gets the lock.
2. It goes into `operator[]`... hasn't exited yet
3. Thread 2 goes to `get` and has no lock. It calls any method in the container. These calls are not thread safe, and so we need to lock the `get` if we are using the mutex to protect this state.

Reply:
I've updated the `get` function to use mutex. Now it should be thread-safe :)